website/source/docs/providers/aws/r/route.html.markdown

```diff
@@ -35,12 +35,14 @@ The following arguments are supported:
 * `destination_cidr_block` - (Required) The destination CIDR block.
 * `vpc_peering_connection_id` - (Optional) An ID of a VPC peering connection.
 * `gateway_id` - (Optional) An ID of a VPC internet gateway or a virtual private gateway.
```

Comment:
@jen20 I'm not super familiar with Go, so please correct me if I'm off base.

On [L121](https://github.com/hashicorp/terraform/pull/4381/files#diff-a9b45e521c3ff848176f0c6492abadabR121), it looks to me like the case for `nat_gateway_id` is identical to the case for `gateway_id` on [L116](https://github.com/hashicorp/terraform/pull/4381/files#diff-a9b45e521c3ff848176f0c6492abadabR116), except for the name of the string.

Is there a call to the AWS Go SDK somewhere which requires specifying which type of gateway it is, or does it just take a generic string identifier for the destination of the route? For example, in the AWS Console UI it just shows route tables as a source and destination and allows the user to pick both NAT Gateways and Internet Gateways from the same dropdown, which is driving my question here.